### PR TITLE
Java 17 features

### PR DESCRIPTION
**Pull request by holgerfriedrich**
- add missing @override
- Java style array syntax
- remove redundant modifiers
- always move String constants to left side in comparisons
- simplify lambda expressions and return statements
- use replace instead of replaceAll w/o regex

### CODE REVIEW

File: bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/ThingDashboardService.java [hunk 1]
```diff
@@ -83,14 +83,14 @@ public ThingDashboardService(NeeoService service, ServiceContext context) {
...
-                || paths[0].equalsIgnoreCase("exportrules") //
-                || paths[0].equalsIgnoreCase("updatedevice"));
+        return paths.length >= 1 && ("thingstatus".equalsIgnoreCase(paths[0]) //
+                || "getchannel".equalsIgnoreCase(paths[0]) //
+                || "getvirtualdevice".equalsIgnoreCase(paths[0]) //
+                || "restoredevice".equalsIgnoreCase(paths[0]) //
+                || "refreshdevice".equalsIgnoreCase(paths[0]) //
+                || "deletedevice".equalsIgnoreCase(paths[0]) //
+                || "exportrules".equalsIgnoreCase(paths[0]) //
+                || "updatedevice".equalsIgnoreCase(paths[0]));
```
[the remark this answers is not in the document]
Comment (wborn) [reply bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/ThingDashboardService.java:93]: Or a `Set<String>` and use `.contains` :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (Hilbrand) [reply bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/ThingDashboardService.java:93]: Then it should be
```suggestion
return paths.length >= 1 && STARTERS.contains(paths[0].toLowerCase(Locale.ROOT));
```

**Review by ccutrer — APPROVED** [review]
LGTM for HomeKit

**Review by lsiepel — APPROVED** [review]
Besides the existing comment : LGTM

**Review by GiviMAD — APPROVED** [review]
LGTM for Porcupineks, Voskstt and ActionTemplateInterpreter.

Thank you for the changes

File: bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/ThingDashboardService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (holgerfriedrich) [reply bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/ThingDashboardService.java]: done

**Review by wborn — APPROVED** [review]
Thanks!

**Pull request merged**